### PR TITLE
return a correct exit code on tests fail

### PR DESCRIPTION
**Pull request by daynin**
It should help to avoid issues like that https://github.com/vishesh/racketscript/issues/195

### CODE REVIEW

File: tests/fixture.rkt [hunk 1]
```diff
@@ -277,7 +284,9 @@
     (displayln "--------------------------------")
     (displayln "::: Optimizations on ::: All :::")
     (displayln "-------------------------------")
-    (run-tests tc-search-patterns)))
+    (set-passed! (run-tests tc-search-patterns)))
+
+  (when (not passed) (exit 1)))
```
Comment (vishesh) [inline tests/fixture.rkt:289]:
```suggestion
  (unless passed (exit 1)))
```

File: tests/fixture.rkt [hunk 2]
```diff
@@ -253,9 +260,9 @@
       (displayln "---------------------------------")
       (displayln "::: Optimizations on ::: none :::")
       (displayln "---------------------------------")
-      (run-tests (filter-not
+      (set-passed! (run-tests (filter-not
```
Comment (vishesh) [inline tests/fixture.rkt:263]: I think this s-exp is not indented after change.

Comment (daynin) [reply tests/fixture.rkt:263]: fixed

**Comment by vishesh** [comment]
Hi @daynin,

Given recent fixes on master, can you rebase this PR against master, so that CI passes before we merge it. Thanks!

**Comment by daynin** [comment]
@vishesh done

**Comment by daynin** [comment]
I think we already can merge with PR because it's only about fixing the testing tool, so it'll help with future work on those errors. @vishesh what do you think?

**Comment by vishesh** [comment]
Oh! My bad. It is fixing the fact failures are showing up in CI report. I mixed up which PR is doing what.

**Pull request merged**